### PR TITLE
docs: update FilterPolicy config to match AWS API

### PR DESCRIPTION
**Pull request by Gadam8**
Currently, filterPolicy configuration outlined in this file does not match what is outlined in the AWS documentation and API - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-subscription.html. The `filterPolicy` should only contain the JSON policy, so no need for the `attributes` property; which in itself was confusing as the policy can apply to the message attributes or body.

Further to this, this definition file was missing the `filterPolicyScope` which is needed to determine whether the user wishes to filter on the message attributes or the body.

**Description**

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: sns/json_schemas/operation.json [hunk 1]
```diff
@@ -97,6 +97,9 @@
         "filterPolicy": {
           "$ref": "#/definitions/filterPolicy"
         },
+        "filterPolicyScope": {
```
Comment (dpwdec) [inline sns/json_schemas/operation.json:100]: As `filterPolicy` and `filterPolicyScope` will be properties *directly* on the object they don't need to be `$ref`'d.

**Review by dpwdec — APPROVED** [review]

**Review by iancooper — DISMISSED** [review]

**Comment by iancooper** [comment]
@dpwdec Approved

**Comment by VisualBean** [comment]
>  whether the user wishes to filter on the message attributes or the body.

sounds odd to me.
The bindings is not for users to create, it for users to consume and learn more about the topic. I dont think consumer configuration should be a part of it.

**Comment by Gadam8** [comment]
> > whether the user wishes to filter on the message attributes or the body.
> 
> sounds odd to me. The bindings is not for users to create, it for users to consume and learn more about the topic. I dont think consumer configuration should be a part of it.
> 
> Are filterpolicies set by the event producer to be used by the consumer?

Although the SNS subscription lives under the SNS remit it is under the control of the consumer of the topic. Filter policies are set by the consumer of the event, not the producer. It is the consumer who is deciding how they wish to consume the events sent by the producer, not the producer determining how their events should be consumed.

**Comment by VisualBean** [comment]
Okay, 
Does it make sense to communicate to consumers of the  specification then? 
I don't immediately see the value in communicating "we filter events based on this" - but maybe I'm missing something

**Review by dpwdec — APPROVED** [review]

**Comment by Gadam8** [comment]
Thanks @dpwdec and @iancooper for the approval. @KhudaDad414 can I get a review please?

**Review by KhudaDad414 — APPROVED** [review]
Schemas seems to match with README.

**Comment by derberg** [comment]
@dpwdec @KhudaDad414 just n info for future -  that when you change structure of binding, you should update binding version too

**Pull request merged**